dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunction.cs

```diff
@@ -156,7 +156,7 @@ internal KernelFunction(string name, string? pluginName, string description, IRe
         Verify.NotNull(kernel);
 
         using var activity = s_activitySource.StartActivity(this.Name);
-        ILogger logger = kernel.LoggerFactory.CreateLogger(this.Name) ?? NullLogger.Instance;
+        ILogger logger = kernel.LoggerFactory.CreateLogger($"{this.GetType()}.{this.Name}") ?? NullLogger.Instance;
```

Comment:
> 1. This means that the implementation detail of the internal derived type's name will end up showing up in logging. That name could easily change. Is this desirable?

I think we want to avoid creating logger with name based on function name, because in this case the number of different loggers created will be equal to number of different functions users execute.

Having `Microsoft.SemanticKernel.KernelFunction` logger name and specific logs like `this._logger.LogInformation("Start function execution: {PluginName}-{FunctionName}", this.PluginName, this.Name");` should be enough, and it will be possible to filter logs by `FunctionName` property in monitoring tool.